src/main/java/reposense/report/ReportGenerator.java

```diff
@@ -59,10 +59,13 @@
             "No authors found with commits for %s (%s).";
     private static final String MESSAGE_START_ANALYSIS = "Analyzing %s (%s)...";
     private static final String MESSAGE_COMPLETE_ANALYSIS = "Analysis of %s (%s) completed!";
+    private static final String FRAGMENT_NUM_REPOS_ANALYZED = "[%d/%d]";
     private static final String MESSAGE_REPORT_GENERATED = "The report is generated at %s";
     private static final String MESSAGE_BRANCH_DOES_NOT_EXIST = "Branch %s does not exist in %s! Analysis terminated.";
 
     private static Date earliestSinceDate = null;
+    private static int totalNumReposToAnalyze = 0;
+    private static int numReposAnalyzed = 0;
```

Comment:
Any reason why `static` variables are preferred here?

As the same can be done using in `for (RepoConfiguration config : configs)` of `ReportGenerator#analyzeRepos`.

Reply:
`ReportGenerator`'s methods in general are used from a static context.

`ReportGenerator#analyzeRepos` only analyses the configs that have the same location. Hence, when we want to analyse configs from multiple locations, `analyzeRepos` will be invoked multiple times and a class-wide variable to keep track of the total number of "repos" (configs) analyzed, `numReposAnalyzed`, is needed.

As for `totalNumReposToAnalyze`, this value is determined from the list of `RepoConfiguration`s, which is found only in `ReportGenerator#generateReposReport`. To access this value in `analyzeRepos`, I thought it is necessary to store it as a class-wide variable.

Reply:
Not sure if its important, but can the implementation play well with tests?

https://travis-ci.org/reposense/RepoSense/jobs/563838015#L1215

Reply:
You should set this to 0 in `generateReposReport` since you have declared this as static variable